File: mobile-customer/lib/haptics.ts
/**
 * Haptic feedback wrapper — اهتزاز خفيف عند الضغط.
 * يصمت بأمان على المحاكي أو الـ web.
 */
import * as Haptics from 'expo-haptics';
import { Platform } from 'react-native';

export const hap = {
  /** عند الضغط على زر عادي */
  tap() {
    if (Platform.OS === 'web') return;
    Haptics.selectionAsync().catch(() => {});
  },
  /** عند ضغطة مهمة (تأكيد، تنفيذ) */
  press() {
    if (Platform.OS === 'web') return;
    Haptics.impactAsync(Haptics.ImpactFeedbackStyle.Medium).catch(() => {});
  },
  /** عند نجاح حدث (طلب أُكّد، حفظ بنجاح) */
  success() {
    if (Platform.OS === 'web') return;
    Haptics.notificationAsync(Haptics.NotificationFeedbackType.Success).catch(() => {});
  },
  /** عند خطأ */
  error() {
    if (Platform.OS === 'web') return;
    Haptics.notificationAsync(Haptics.NotificationFeedbackType.Error).catch(() => {});
  },
  /** تحذير */
  warning() {
    if (Platform.OS === 'web') return;
    Haptics.notificationAsync(Haptics.NotificationFeedbackType.Warning).catch(() => {});
  },
};

File: mobile-flutter/packages/daari_core/lib/src/services/haptics.dart
import 'package:flutter/services.dart';

/// غلاف الاهتزاز اللمسي — منقول من `lib/haptics.ts` في Expo.
///
/// دوال `HapticFeedback` تصمت بأمان على الأجهزة/المنصّات غير الداعمة، فلا حاجة
/// لفحص المنصّة. لا ننتظر الـ Future (إطلاق وننسى).
class Hap {
  Hap._();

  /// ضغطة عادية (تحديد/تنقّل).
  static void tap() => HapticFeedback.selectionClick();

  /// ضغطة مهمة (تأكيد، تنفيذ إجراء).
  static void press() => HapticFeedback.mediumImpact();

  /// نجاح حدث (طلب أُرسل، حفظ ناجح). Flutter بلا نوع «نجاح» مخصّص → نبضة متوسطة.
  static void success() => HapticFeedback.mediumImpact();

  /// خطأ — نبضة قويّة.
  static void error() => HapticFeedback.heavyImpact();

  /// تحذير — اهتزاز قصير.
  static void warning() => HapticFeedback.vibrate();
}
